ERCS/erc-7995.md

```diff
@@ -0,0 +1,587 @@
+
+### Type Safety and Interoperability
+
+All operations **MUST** preserve type safety by accepting and returning specific encrypted data types. Mixing types (e.g., euint32 with euint64) **MUST** be explicitly cast via well-defined conversion functions if such behavior is desired. This avoids ambiguity and enforces consistency across contracts and toolchains.
+
+Constraints and Considerations
+
+- Operations **MUST NOT** leak information about the encrypted operands or results.
+- Operations **SHOULD** be implemented in a way that is compatible with circuit-based or
+  homomorphic evaluation strategies.
+- Implementers **MAY** introduce operation limits (e.g., maximum supported depth or
+  operand size) depending on performance or cryptographic constraints.
+
+This module allows developers to perform meaningful computation on private data while maintaining the composability and auditability guarantees of Ethereum smart contracts.
+
+### On-Chain ACL for Handle-Based Permissions
+
+The ACL contract **MAY** provide functions to allow a `bytes32` value for a given address, and to check
+whether such a value is allowed.
+
+```solidity
+function allow(bytes32 handle, address entity) external;
+function isAllowed(bytes32 handle, address entity) external view
+returns (bool);
+```
+
+- allow: Grants access for a given address to a specific handle.
+- isAllowed: Returns true if the address has access to the given handle.
+
+The ACL **MAY** include account delegation:
+
+```solidity
+function delegateAccount(address delegatee, address[] memory contractAddresses, uint expiration) external;
+function revokeDelegation(address delegatee, address[] memory contractAddresses) external;
+function allowedOnBehalf(address delegatee,bytes32 handle,address contractAddress,address account) external view returns (bool);
+```
+
+- delegateAccount: Grants delegation rights to delegatee, allowing them to act on behalf of the sender for the specified contract addresses with an expiration date (timestamp). This delegation **MAY** apply to all handles associated with the sender within those contracts.
+- revokeDelegation: Revokes previously granted delegation rights from delegatee for the specified contract addresses.
+- allowedOnBehalf: Returns true if delegatee is authorized to act on behalf of account for the given handle and contract address; otherwise returns false.
+
+Smart contracts can integrate this module to perform permission checks before granting access to sensitive operations.
+
+### JavaScript API
+
+To support encrypted data workflows in frontend applications, a reference JavaScript library provides a standardized interface for constructing encrypted inputs, validating decryption access, and sending decryption requests—either directly or via delegation, based on cryptographic proofs using EIP-712.
```

Comment:
I think this section should be moved to the Reference Implementation section, since this is all non-normative information.